moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -339,9 +343,8 @@ bool RuckigSmoothing::runRuckig(robot_trajectory::RobotTrajectory& trajectory,
           }
           target_state->update();
         }
-        ruckig_ptr = std::make_unique<ruckig::Ruckig<ruckig::DynamicDOFs>>(num_dof, timestep);
         initializeRuckigState(*trajectory.getFirstWayPointPtr(), group, ruckig_input, ruckig_output);
-        // Begin the while() loop again
+        // Begin the for() loop again
```

Comment:
For readability, it would make a lot of sense to break the code into separate functions to loop over. There are so many for loops in here.

Reply:
Did a first pass at this with a 100-line refactoring

Reply:
sweet, already helps a lot